midi_app_controller/gui/midi_status.py

```diff
@@ -182,8 +186,18 @@ def _select_controller(self, controller: Optional[SelectedItem]) -> None:
         state = get_state_manager()
         state.select_controller(controller)
         state.select_recent_binds()
+        self.select_recent_midi_ports()
         self.refresh()
 
+    def select_recent_midi_ports(self):
+        """Selects recent MIDI ports for the current controller."""
+        state = get_state_manager()
+        state.select_recent_midi_ports()
+        self.current_midi_in.refresh_items()
+        self.current_midi_in.set_current(state.selected_midi_in)
+        self.current_midi_out.refresh_items()
+        self.current_midi_out.set_current(state.selected_midi_out)
+
```

Comment:
This method is not needed. Just call `state.select_recent_midi_ports()` in `_select_controller()` just like `state.select_recent_binds()` is called. `self.refresh()` is called in the last line of `_select_controller()`, which updates the values.